### PR TITLE
Add rw typeCheck command

### LINKED ISSUE #2788: Error: Cannot find module 'types/graphql' after upgrade to v0.33.0
Getting this error after upgrading to v0.33.0 when I try to import enums from src/(api/web)/types/graphql.d.ts. Before upgrading, I used 'graphql-code-generator' package but configured it to generate regular '.ts' files and of course importing enums worked just fine. I see in docs for graphql-code-generator that the 'constEnums' config option for the 'typescript' plugin would probably fix this I think. See in the docs: [constEnums](https://www.graphql-code-generator.com/docs/plugins/typescript)

### LINKED ISSUE #2797: Add `rw typecheck` command to run tsc
Currently redwood's Typescript support is essentially used as linting. There are no build time checks, or an easy way to check if your TypeScript project is a-ok!

Command to run: 
`yarn rw type-check`
maybe alias `yarn rw tsc`

### LINKED ISSUE #2811: Expose types from Cell when they're used
We're not currently generating types for when we _use_ Cells. i.e. the Cell input props, which are used to make the graphql query.

### PR DESCRIPTION
**Pull request by dac09**
## What does it do?

- Adds `rw typecheck` command, that allows users to run a typecheck 
Fixes #2797
- Cell generator runs `rw gen` after creating a Cell, incase you don't have dev server running.
- Updating now passed to failure component
- adds the ability to pass `CellInput` type to CellSuccessProps  as a generic. This is a nice to have, because it makes the code look cleaner

I've committed things in an organised way so that it's easier to see changes commit-by-commit.

### CODE REVIEW

**Comment by dac09** [comment]
@jtoar I've made a small change to the updating param, would you mind taking a look please? 

My assumptions:
a) `updating` is made available to Success, Empty and Loading, but not to Failure
b) doesn't matter what query library you're using (i.e. apollo or react-query) updating will always be present to the above components

**Comment by jtoar** [comment]
@dac09 

> a) updating is made available to Success, Empty and Loading, but not to Failure

`updating` shouldn't be available to Loading since that'd be redundant (`updating` is derived from the `loading` boolean—it's just `loading` renamed). I didn't pass `updating` to Failure in my original PR, but I think it actually makes sense to do so—a query can fail and be retried. What do you think?

> b) doesn't matter what query library you're using (i.e. apollo or react-query) updating will always be present to the above components

Yes!

I'm also seeing an error by one of the actions I can look into:

```
'updating' is specified more than once, so this usage will be overwritten.
```

**Comment by peterp** [comment]
@dac09 I added some information to the original issue about typing this in the mirror Cell file, since that would also work really nicely for JS projects.

**Comment by dac09** [comment]
Updating this PR to _only_ contain changes for `rw typecheck` command. Will move the other things to a separate PR

**Review by peterp — APPROVED** [review]
The typecheck stuff looks good, I'm assuming you'll remove the cell stuff?

**Comment by dac09** [comment]
> The typecheck stuff looks good, I'm assuming you'll remove the cell stuff?

The cell stuff is actually valid changes Peter. The description was updated, so all the changes described there are valid

**Comment by peterp** [comment]
Ah ok, you're last comment was:
> Updating this PR to only contain changes for rw typecheck command, and the minor cell changes. Will move the other things to a separate PR

So I was confused about seeing the cell stuff in here.

File: packages/cli/src/commands/generate/cell/cell.js [hunk 1]
```diff
@@ -169,4 +171,14 @@ export const { command, description, builder, handler } =
...
+    includeAdditionalTasks: () => {
+      return [
+        {
+          title: `Generating types ...`,
+          task: async () => {
+            return generate()
+          },
+        },
+      ]
+    },
```
Comment (peterp) [inline packages/cli/src/commands/generate/cell/cell.js:183]: 🤔 Should we make all generators just generate types as well?

File: packages/web/src/components/createCell.tsx [hunk 2]
```diff
@@ -128,7 +129,13 @@ export function createCell<CellProps = any>({
         {({ error, loading, data, ...queryRest }) => {
           if (error) {
             if (Failure) {
-              return <Failure error={error} {...queryRest} {...props} />
+              return (
+                <Failure
+                  error={error}
+                  {...{ updating: loading, ...queryRest }}
+                  {...props}
```
Comment (peterp) [inline packages/web/src/components/createCell.tsx:136]: Maybe something like this? Seems a bit weird to spread for 1, but not the other? 🤷‍♂️
```suggestion
                  {...{ updating: loading, ...queryRest, ...props }}
```

**Review by peterp — APPROVED** [review]
Still looks good to me.

File: packages/web/src/components/createCell.tsx (hunk 2 above)
Comment (dac09) [reply packages/web/src/components/createCell.tsx:136]: yeah I just copied what was done below lol. This does look better

Comment (dac09) [reply packages/web/src/components/createCell.tsx:136]: I'll clean this up the the cell typing PR!

**Comment by peterp** [comment]
Oh lol, I actually just re-read your comment...! I had read the email earlier which said it was just the type-check command, but I saw that you edited it in the meantime! 🤦‍♂️

**Pull request merged**